### PR TITLE
Feature/emergency additional tab

### PR DESCRIPTION
**Pull request by shreeyash07**
## Addresses:
-  [Emergency Additional Tab](https://github.com/orgs/toggle-corp/projects/8/views/21?pane=issue&itemId=29928562)

## Changes
- Add dynamic additional tab in emergencies page

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/Emergency/index.tsx [hunk 1]
```diff
@@ -116,6 +166,18 @@ export function Component() {
...
+                    <NavigationTab
+                        to="emergencyAdditionalTab"
+                        urlParams={{
+                            routeName: tab.routeName,
+                            emergencyId,
+                        }}
+                    >
+                        {tab.name}
+                    </NavigationTab>
+                ))}
```
Comment (samshara) [inline src/views/Emergency/index.tsx]: we need to pass `key` prop here.

File: src/views/EmergencyAdditionalTab/index.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    return (
+        <div
+            className={styles.additionalTab}
+        >
+            {additionalTab?.snippets?.map((sanitize) => (
+                <HtmlOutput
+                    value={sanitize.snippet}
+                    className={styles.additionalContent}
+                />
+            ))}
```
Comment (samshara) [inline src/views/EmergencyAdditionalTab/index.tsx]: we need to pass `key` prop here as well

File: src/views/Emergency/index.tsx [hunk 3]
```diff
@@ -19,13 +19,19 @@ import { type EmergencyOutletContext } from '#utils/outletContext';
 import i18n from './i18n.json';
 import styles from './styles.module.css';
 
+function routeName(text: string | null | undefined) {
```
Comment (tnagorra) [inline src/views/Emergency/index.tsx]:
```suggestion
function getRouteIdFromName(text: string) {
```

File: src/views/EmergencyAdditionalTab/index.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+export function Component() {
+    const { routeName } = useParams<{ routeName: string }>();
+    const { emergencyAdditionalTabs } = useOutletContext<EmergencyOutletContext>();
+
+    const additionalTab = emergencyAdditionalTabs?.find((tab) => tab.routeName === routeName);
+    return (
+        <div
+            className={styles.additionalTab}
+        >
+            {additionalTab?.snippets?.map((sanitize) => (
```
Comment (tnagorra) [inline src/views/EmergencyAdditionalTab/index.tsx]:
```suggestion
            {additionalTab?.snippets?.map((snippet) => (
```

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Pull request merged**